app/controllers/api/event_streams_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Api
+  class EventStreamsController < BaseController
+    before_action :validate_filters, :ensure_pagination, :only => :index
+
+    private
+
+    def validate_filters
+      messages = []
+      messages << "must specify target_type" unless filter_contains?("target_type")
+      messages << "must specify a minimum value for timestamp" unless filter_contains?("timestamp>")
+      raise BadRequestError, messages.join(", ").capitalize if messages.any?
+    end
+
+    def filter_contains?(filter)
+      Array(params["filter"]).any? { |f| f.start_with?(filter) }
+    end
+
+    def ensure_pagination
+      params["limit"] ||= Settings.api.metrics_default_limit
```

Comment:
I'm leaning on this configuration rather than creating a new one. @jntullo WDYT about renaming to `large_collection_default_limit` or something similar?

Reply:
@imtayadeway I like it! 👍